File: java_test/src/main/java/com/juice/java_test/TestString.java
package com.juice.java_test;

/**
 * @author juice
 * @desc String 知识点
 */
public class TestString {

    public static void main(String[] args) {

        testEquals("asd", "asd");
        testEquals("asD", "Asd");
    }

    /**
     * @see java.lang.String#equalsIgnoreCase
     * usage: checking vertification code
     */
    private static void testEquals(String a, String b) {

        System.out.println("\"" + a + "\".equals(\"" + b + "\") = " + a.equals(b));
        System.out.println("\"" + a + "\".equalsIgnoreCase(\"" + b + "\") = " + a.equalsIgnoreCase(b));
    }
}

File: kotlin_test/src/main/java/com/juice/kotlin_test/TestString.kt
package com.juice.kotlin_test

/**
 * @author juice
 * @date 2020/12/10
 * @desc test for String
 */
class TestString {

    companion object {

        @JvmStatic
        fun main(args: Array<String>) {

            testEquals("asd", "asd")
            testEquals("asD", "Asd")
        }

        private fun testEquals(a: String, b: String) {

            println("\"$a\".equals(\"$b\") = " + (a == b))
            println("\"$a\".equalsIgnoreCase(\"$b\") = " + a.equals(b, ignoreCase = true))
        }
    }
}